src/udp.c

```diff
@@ -41,6 +43,19 @@ enftun_udp_connect_addr(struct enftun_udp* udp, struct sockaddr* connect_addr)
         return -1;
     }
 
+    if (mark > 0)
+    {
+        if ((rc = setsockopt(udp->fd, SOL_SOCKET, SO_MARK, &mark,
+                             sizeof(mark))) < 0)
+        {
+            enftun_log_error("UDP: Failed to set mark %d: %s\n", mark,
+                             strerror(errno));
+            rc = -errno;
+            close(udp->fd);
```

Comment:
I didn't notice it before, but let's refactor all the `close(udp->fd)` calls into a section at the bottom of this function.  Similar to [here](https://github.com/xaptum/enftun/blob/94ccb214ae466c045c7f84e631b81292833be582/src/tcp.c#L116).

Also, set `udp->fd` to 0 so that we don't accidentally try to use it again. (`-1` would be safer, since file descriptor 0 is stdin.  But the rest of the code using `0`, so let's stick with that.  Changing those to `-1` is a future task.)